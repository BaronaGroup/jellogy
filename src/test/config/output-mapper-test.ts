import {prepareTestKarhu} from '../test-utils'

describe('output-mapper-test', () => {
  describe('it is applied', () => {
    const karhuTest = prepareTestKarhu({
      outputMapper: val => val * 2
    })

    it('is applied to the first parameter', karhuTest((karhu, output) => {
      karhu.context('ctx').error(10000 / 2)
      expect(output.tracked).toMatchSnapshot()
    }))

    it('is applied to all parameters', karhuTest((karhu, output) => {
      karhu.context('ctx').error(1, 2, 3, 4, 5, 6)
      expect(output.tracked).toMatchSnapshot()
    }))
  })

  describe('args', () => {
    const args: any[] = []

    const karhuTest = prepareTestKarhu({
      outputMapper: (...argsImpl: any[]) => {
        args.push(argsImpl)
        return 'Overridden'
      }
    })

    it('is called with the right arguments', karhuTest((karhu) => {
      karhu.context('test-context').warn('Hello', 1, {a: 1})
      expect(args).toMatchSnapshot()
    }))
  })

  describe('transport', () => {
    const karhuTest = prepareTestKarhu({
        outputMapper: val => val * 2
      },
      {
        outputMapper: val => val + 10
      }
    )

    it('can provide its own mapper', karhuTest((karhu, output) => {
      karhu.context('ctx').error(5)
      expect(output.tracked).toMatchSnapshot()
    }))
  })
})
