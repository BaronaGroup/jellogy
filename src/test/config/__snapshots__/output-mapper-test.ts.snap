// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`output-mapper-test args is called with the right arguments 1`] = `
Array [
  Array [
    "Hello",
    "WARN",
    "test-context",
    Array [
      "Hello",
      1,
      Object {
        "a": 1,
      },
    ],
    0,
  ],
  Array [
    1,
    "WARN",
    "test-context",
    Array [
      "Hello",
      1,
      Object {
        "a": 1,
      },
    ],
    1,
  ],
  Array [
    Object {
      "a": 1,
    },
    "WARN",
    "test-context",
    Array [
      "Hello",
      1,
      Object {
        "a": 1,
      },
    ],
    2,
  ],
]
`;

exports[`output-mapper-test it is applied is applied to all parameters 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [
    Array [
      "[1] [101m[97mERROR ctx 2 4 6 8 10 12[39m[49m",
      "ERROR",
      "ctx",
    ],
  ],
}
`;

exports[`output-mapper-test it is applied is applied to the first parameter 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [
    Array [
      "[1] [97m[101mERROR ctx 10000[49m[39m",
      "ERROR",
      "ctx",
    ],
  ],
}
`;

exports[`output-mapper-test transport can provide its own mapper 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [
    Array [
      "[1] [97m[101mERROR ctx 15[49m[39m",
      "ERROR",
      "ctx",
    ],
  ],
}
`;
